aspnetcore/fundamentals/openapi/openapi-comments.md

```diff
@@ -0,0 +1,307 @@
+
+The `MemberKey` definition attempts to encode as much information as possible to map a compile-time symbol to its runtime counterpart.
+
+```csharp
+internal sealed record MemberKey(
+    string? DeclaringType,
+    MemberType MemberKind,
+    string? Name,
+    string? ReturnType,
+    string[]? Parameters) : IEquatable<MemberKey>
+```
+
+### Code Generation
+
+The generator emits code that contains:
+
+* A cache of XML comments mapped to member identifiers:
+   <!-- REVIEW: In additon to the following, provide a concret example -->
+   ```csharp
+   _cache.Add(new MemberKey(/*...*/), new XmlComment(/*...*/));
+   ```
+* OpenAPI transformer implementations:
+   * [`XmlCommentOperationTransformer`](https://github.com/dotnet/aspnetcore/blob/main/src/OpenApi/gen/XmlCommentGenerator.Emitter.cs#L229):  Applies comments to API operations (methods).
+   * [`XmlCommentSchemaTransformer`](https://github.com/dotnet/aspnetcore/blob/main/src/OpenApi/gen/XmlCommentGenerator.Emitter.cs#L308): Applies comments to data models (types).
+* Extension methods that intercept <xref:Microsoft.Extensions.DependencyInjection.OpenApiServiceCollectionExtensions.AddOpenApi*> calls to inject the transformers:
+
+   ```csharp
+   public static IServiceCollection AddOpenApi(this IServiceCollection services)
+   {
+       return services.AddOpenApi("v1", options =>
+       {
+           options.AddSchemaTransformer(new XmlCommentSchemaTransformer());
+           options.AddOperationTransformer(new XmlCommentOperationTransformer());
+       });
+   }
+   ```
+
+### Interception mechanism
+
+The generator uses the C# compiler's interceptor feature to intercept calls to the `<xref:Microsoft.Extensions.DependencyInjection.OpenApiServiceCollectionExtensions.AddOpenApi*> method. The generator:
+
+1. Detects different `AddOpenApi` overloads.
+2. Generates appropriate interceptor methods for each overload.
+3. Adds the XML comment transformers to the OpenAPI options.
+
+When the interceptor runs, it:
+
+1. Adds the XML-specific transformers.
+1. Applies any transformers the user has registered in their application code. This allows user transformers to inspect metadata that has been automatically set by the source generated transformer implementations.
+
+### Runtime behavior
+
+When the generated code runs:
+
+1. The XML comment cache is populated on first use with structured comment data.
+2. The intercepted `AddOpenApi` methods add the transformers to the OpenAPI options.
+3. During API documentation generation, the transformers:
+   * Look up documentation for API methods and apply summaries, descriptions, etc.
+   * Apply parameter documentation to OpenAPI parameters.
+   * Set response descriptions based on XML documentation.
+   * Include [examples](#add-examples) when provided in the XML.
+
+The generator processes standard XML code comments and adds them to the generated OpenAPI documentation.
+
+## Frequently Asked Questions
```

Comment:
Are FAQs common in our docs? I don't think any of the other OpenAPI docs have a section like this. Should we have a separate FAQ covering all of OpenAPI generation and move these there?